### ISSUE TITLE
Steam (non-beta) Flatpak crash on startup: Error "steamwebhelper is not responding" - NixOS 24.05.

### ISSUE THREAD

**Issue by shymega** [flathub/com.valvesoftware.Steam#1347]
Similar issue to #1318, except on NixOS 24.05, with the latest Flatpak.

I'm using this on an AMD CPU, with an AMD 780M iGPU.

The fix described [here](https://github.com/ValveSoftware/steam-for-linux/issues/11006#issuecomment-2196024433) works for me, but it does mean I can't use the latest version of Steam.

I do not have AppArmor or SELinux on NixOS, and my Flatpak has no special overrides, other than allowing access to external storage for games.

I'm not sure where to find the logs - if someone can let me know where to find the logs, I can upload them.

Thanks.

**Comment by smcv** [comment]
> Similar issue to https://github.com/flathub/com.valvesoftware.Steam/issues/1318

I'm pretty sure you mean a similar *symptom*, but almost certainly for a different reason.

> [forcibly downgrading Steam] works for me

This is not something that Valve is ever going to support.

> I'm not sure where to find the logs

`~/.var/app/com.valvesoftware.Steam/.local/share/Steam/logs`

**Comment by shymega** [comment]
> > Similar issue to #1318
> 
> I'm pretty sure you mean a similar _symptom_, but almost certainly for a different reason.

Yes, you're quite right. My bad for neglecting that.
> 
> > [forcibly downgrading Steam] works for me
> 
> This is not something that Valve is ever going to support.

Agreed, which is why I would rather not use that [solution] as a fix.
> 
> > I'm not sure where to find the logs
> 
> `~/.var/app/com.valvesoftware.Steam/.local/share/Steam/logs`

Any log in particular? I see quite a few.

**Comment by smcv** [comment]
`console-linux.txt`, `webhelper-linux.txt` and `webhelper.txt` are likely to be the most relevant for a steamwebhelper crash. If you revert your workaround(s), move the current logs out of the way, then attempt to start Steam, you'll get "clean" logs that only contain the most recent launch attempt.

**Comment by shymega** [comment]
Got logs from the default Steam version.

I couldn't get `webhelper.txt` - it doesn't exist - but the rest I have.

Attached.
[console-linux.txt](https://github.com/user-attachments/files/17067081/console-linux.txt)
[webhelper-linux.txt](https://github.com/user-attachments/files/17067082/webhelper-linux.txt)

**Comment by smcv** [comment]
Try with `STEAM_LINUX_RUNTIME_VERBOSE=1` in the environment?

If you have `STEAM_LINUX_RUNTIME_LOG=1` set, there will be another log file in `~/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_64/steam-runtime-sniper/var/`.

**Comment by smcv** [comment]
> I couldn't get `webhelper.txt` - it doesn't exist

That would be consistent with container runtime startup freezing or something, so that the actual `steamwebhelper` never actually starts.

**Comment by shymega** [comment]
OK, so when running with those env vars (cmd: ` flatpak run --env=STEAM_LINUX_RUNTIME_LOG=1 --env=STEAM_LINUX_RUNTIME_VERBOSE=1 com.valvesoftware.Steam`), Steam does load.

However, it appears to be intermittent. I have attached the logs requested to this ticket. I'm not sure if it's related to a possible bug with NixOS and AMD - its affecting 1Password GUI and `dwl`, a `wlroots` WM for me at the moment.

[console-linux.txt](https://github.com/user-attachments/files/17080889/console-linux.txt)
[slr-latest.log](https://github.com/user-attachments/files/17080890/slr-latest.log)
[webhelper.txt](https://github.com/user-attachments/files/17080891/webhelper.txt)
[webhelper-linux.txt](https://github.com/user-attachments/files/17080892/webhelper-linux.txt)

**Comment by smcv** [comment]
If this is failing intermittently, please continue to use those options for a while (you can use `flatpak override --user com.valvesoftware.Steam` to get those two environment variables set by default for Steam, if you want) and collect logs from one of the times that it fails.

**Comment by shymega** [comment]
It's happened consistently now. I've got four logs from Flatpak, and three from Steam running natively on my distro (NixOS).


I am curious if this bug I am experiencing is related: https://github.com/NixOS/nixpkgs/issues/343806

However, that bug is specific to mixing unstable with stable NixOS channels in relation to Mesa versions.

I'm also wondering if it's worth reporting this to upstream Steam. What do you think? I've attached logs to this comment.

Running Steam natively crashes before the login screen, and I have not been able to get both 'webhelper' logs - the only one I can get is 'webhelper-linux.txt'. The crash happens after unpacking the Steam runtime. The `slr-latest.log` file with native Steam is also non-existent.

Flatpak Steam logs:

[console-linux-flatpak.txt](https://github.com/user-attachments/files/17163067/console-linux-flatpak.txt)
[slr-latest-flatpak.log](https://github.com/user-attachments/files/17163068/slr-latest-flatpak.log)
[webhelper-flatpak.txt](https://github.com/user-attachments/files/17163069/webhelper-flatpak.txt)
[webhelper-linux-flatpak.txt](https://github.com/user-attachments/files/17163070/webhelper-linux-flatpak.txt)

Native Steam logs:

[console-linux-native.txt](https://github.com/user-attachments/files/17163135/console-linux-native.txt)
[webhelper-linux-native.txt](https://github.com/user-attachments/files/17163136/webhelper-linux-native.txt)

**Comment by emmetoneillpdx** [comment]
I'm also running into a "steamwebhelper is not responding" crash on my system (Bluefin 40  (Fedora Silverblue 40), Ryzen 5900x, RX 590, 64GB system RAM).

Crash was infrequent at first and fixable by killing the webhelper process, but now I can't seem to get Steam to boot at all.

[console-linux.txt](https://github.com/user-attachments/files/17239163/console-linux.txt)
[webhelper-linux.txt](https://github.com/user-attachments/files/17239159/webhelper-linux.txt)

**Comment by emmetoneillpdx** [comment]
Same "steamwebhelper is not responding" issue here.

Was originally running into this issue infrequently and was able to resolve it by killing the webhelper process or restarting my session. But starting today I've been running into it repeatedly and can no longer open Steam, which sucks.

**My System:**

Bluefin 40 (Fedora Silverblue 40)
Gnome 46 x11
Ryzen 9 5900x
RX 590
64 GB RAM
Steam Flatpak via Flathub  (com.valvesoftware.Steam) 

**Logs:**

[console_log.txt](https://github.com/user-attachments/files/17239530/console_log.txt)
[console-linux.txt](https://github.com/user-attachments/files/17239531/console-linux.txt)
[webhelper-linux.txt](https://github.com/user-attachments/files/17239532/webhelper-linux.txt)
[steamui.txt](https://github.com/user-attachments/files/17239533/steamui.txt)

**Comment by lbrunkho** [comment]
I started seeing this today as well. Looks like steam rolled out a runtime update. Running Fedora 40. 

I see this line of interest in my webhelper-linux.txt...

`[2024-10-02 22:20:20] ./steamwebhelper: error while loading shared libraries: ./libcef.so: cannot allocate memory in static TLS block`

**Comment by lbrunkho** [comment]
Ran my steam with `-clearbeta` to roll back the steam runtime and that seemed to fix it for me. If it is still broken tomorrow on the beta branch, then I'll report the error upstream.

**Comment by jpkhawam** [comment]
Wanted to mention that I also faced this but only after the October 3rd beta update, I was on the beta branch before just fine. I forgot to check this issue tracker and logged an issue upstream here: https://github.com/ValveSoftware/steam-for-linux/issues/11312

Should I delete or keep it?

**Comment by shymega** [comment]
I don't think this is the same issue - I'm not using the Beta branch. As well as that, this issue is scoped to NixOS. I would advise opening a new issue for your issues.

**Comment by izmyname** [comment]
It just got broken for me 15 minutes ago, after some 'linux unpacking runtime environment'

**Comment by bbhtt** [comment]
Steam most likely needs to backport https://github.com/chromiumembedded/cef/commit/be4085981cf2035576dee64813ace9d6b1930305 or update to a fixed libcef build.

**Comment by RobotRoss** [comment]
Also seeing this, Fedora 40 Wayland.

**Comment by shymega** [comment]
It's getting a bit noisy with "Me too!" comments. Can new reports please read through the issue, for the first comment that matches your configuration. For example, in the case of Fedora 40/Wayland, you would then react [to the comment] with 👍🏻 (`:+1:`), and subscribe to the issue. If there is no comment that matches your setup (i.e., distro, and windowing system: X11 or Wayland), then please comment with that info.

It's also worth stressing this tracker is for the Steam Flatpak - *not* for the normal/native Steam application.

This sounds like a bug that's affecting quite a few people. Let's keep the issue tidy, and focused on the resolution - otherwise, it just creates noise, and the fix will be delayed, or worse, not happen at all.

**Comment by orpheus333** [comment]
I'm seeing this on a intel i7 3720QM, HD4000 graphics running Ubuntu 24.04.1 LTS. Steam is installed using the 1.0.0.81 Flatpak. The stable version of steam seems to work fine without any steamwebhelper issues. However when updating to the beta client I get the bug and steam crashes on launch with the webhelper issue. Using -clearbeta to rollback to the stable client fixes the issue. So i'm not sure if this is specifically a Flatpak problem.

**Comment by shymega** [comment]
This isn't related purely to beta. It happens on both beta+stable in Flatpak or native Steam.

**Comment by shymega** [comment]
The issue should be renamed, I'm running into the crash on stable Steam channel, with Flatpak & native (NixOS).

**Comment by smcv** [comment]
@shymega, are you seeing the message "libcef.so: cannot allocate memory in static TLS block" in the log?

If you are not, then you and everyone else here are probably seeing two different crashes. Unfortunately, "Steamwebhelper is not responding" is not a distinctive error condition - it's a symptom, not a cause - so it's very easy to get people who are experiencing similar crashes for different reasons jumping onto an issue report and derailing it.

For anyone who is seeing the message "libcef.so: cannot allocate memory in static TLS block" in the beta, the relevant Steam issue report is https://github.com/ValveSoftware/steam-for-linux/issues/11312. Please subscribe there and wait for a Valve developer to act on it. The workaround is to downgrade from beta to stable (which you can't easily do from the GUI because it isn't starting, but `flatpak run com.valvesoftware.Steam -clearbeta` or deleting the file `~/.var/app/com.valvesoftware.Steam/.steam/root/package/beta` might help).

@shymega, if you're seeing a different error message or no error message, then that should be reported to Valve via https://github.com/ValveSoftware/steam-for-linux/issues separately. You can make your report as simple and actionable as possible for Valve developers (therefore maximizing the likelihood that someone can fix it) by reporting it as an issue with the non-Flatpak, non-sandboxed ("native") version of Steam (because that's a simpler scenario than the Flatpak app), and providing logs - at least `console-linux.txt` and `webhelper-linux.txt`, but preferably a tar archive of the complete logs directory as requested in the steam-for-linux issue template.

**Comment by shymega** [comment]
@smcv I am _not_ seeing the `libcef.so` error. Let us rename this issue, and point people encountering the `libcef.so` error to a separate issue.

My problem happens on both native and Flatpak Steam, so I think we should keep this issue open and refocus on the original problem reported. Then, I can link this issue to upstream Steam as well.

What do you think?

**Comment by smcv** [comment]
> @smcv I am _not_ seeing the `libcef.so` error. Let us rename this issue, and point people encountering the `libcef.so` error to a separate issue.

Yes, I think that would be best. (But I am not a maintainer of this project, so I cannot retitle the issue for you.)

**Comment by shymega** [comment]
@smcv I've done it now.

**Comment by shymega** [comment]
I'm still experiencing the crash, so I'm going to report it upstream, and link to this issue. I suspect once fixed, it'll resolve both Flatpak and native.

@bbhtt - could you 'hide' the `libcef.so` comments as off-topic, given my original post was not related to `libcef.so`? Thanks.

**Comment by shymega** [comment]
Attached fresh logs: 
[steam-logs-flatpak.tar.gz](https://github.com/user-attachments/files/17356878/steam-logs-flatpak.tar.gz)
[steam-logs-native.tar.gz](https://github.com/user-attachments/files/17356879/steam-logs-native.tar.gz)

And a screenshot of the webhelper dialog:

![steamwebhelper-dialog](https://github.com/user-attachments/assets/ba65d0f7-4d79-4abb-ab3f-d767be9b5540)

Linked issue on upstream: https://github.com/ValveSoftware/steam-for-linux/issues/11346

**Comment by bbhtt** [comment]
> @bbhtt - could you 'hide' the libcef.so comments as off-topic, given my original post was not related to libcef.so? Thanks.

can you please open a new issue? Everyone else's logs are related to libcef. It is better to keep this bug focused on that particular issue.

**Comment by shymega** [comment]
On 13.10.2024 18:28, bbhtt wrote:
>> @bbhtt - could you 'hide' the libcef.so comments as off-topic, given my original post was not related to libcef.so? Thanks.
>
>can you please open a new issue? Everyone else's logs are related to libcef. It is better to keep this bug focused on that particular issue.

@bbhtt I don't think that's necessary. I've already linked to it on
upstream Steam, and, to an extent, this issue has been sidelined by
unrelated bug reports.

I'd rather we just hide the unrelated comments, and open a new tracking
issue for `libcef.so` if necessary.

Best wishes,
--
Dom Rodriguez
GPG Fingerprint: EB0D 45E6 D0DC 1BA1 A2B5  FC24 72DC F123 1E54 BD43

**Comment by shymega** [comment]
I did wonder if this was related to a NixOS bug with mixing Mesa between stable (at the time, 24.05) and unstable channels, but 24.11 has now been released, and this bug is still present.

**Comment by shymega** [comment]
Seems to be resolved in the latest release of Steam. No longer reproducible as of last week. Closing as resolved.
